### ISSUE TITLE
FFS: Why is tf/keras no longer seeing the GPU?

### ISSUE THREAD

**Issue by SteveOv** [SteveOv/ebop_maven#10]

**Comment by SteveOv** [comment]
ebop_maven conda env installs cudnn=8.1

will try an upgrade.

**Comment by SteveOv** [comment]
conda upgrade cudnn installs 8.9.2.26 but doesn't fix the problem

**Comment by SteveOv** [comment]
Just confirmed my old code on tf/keras 2.6 can see the GPU fine.

**Comment by SteveOv** [comment]
FFS - broken install of CUDA toolkit!  I'm off to bed.

**Comment by SteveOv** [comment]
I still had cudnn pinned at version 8.1 in the ebop_maven conda env. The [comatibility list](https://www.tensorflow.org/install/source#gpu) states it  should be at 8.9 for the tf 2.16 and Cuda 12.3.

I'll remove the restriction and recreate the env to see what happens.

**Comment by SteveOv** [comment]
Env sets up OK with cudnn 8.9 and Cuda 12.4 but still the training refuses to see the GPU.

**Comment by SteveOv** [comment]
Looks like it may be an issues with tf v2.16.1

See https://github.com/tensorflow/tensorflow/issues/63362#issuecomment-2016019354

**Comment by SteveOv** [comment]
Can confirm the temporary fix above works. This isn't a blocker for me so I'm happy to wait on newer versions of tensorflow. In the meantime I've found that installing tensorflow[and-cuda] removes the need for an explicit cudnn install so I'm going to update the environment.yaml/requirements.txt to that effect.

**Comment by SteveOv** [comment]
Won't fix - it's an issue with tensorflow 2.16.0 release and workarounds exist if needed.
